py3status/modules/emerge_status.py

```diff
@@ -0,0 +1,146 @@
+# -*- coding: utf-8 -*-
+"""
+Emerge_status provides a short information on the current
+emerge status if there is an emerge process currently running
+
+Configuration parameters:
+    cache_timeout: how often we refresh this module in second (default 1)
+    emerge_log_file: in case it is placed somewhere else than the default
+        (default '/var/log/emerge.log')
+    format: Display format to user (default '{current} / {total}')
+    hide_if_zero: Don't show in bar if there is no emerge running (default True)
+
+Format placeholders:
+    {current} Number of package that is currently emerged
+    {total} Total number of packages that will be emerged
+    {category} Category of the currently emerged package
+    {pkg} Name of the currently emerged packaged
+
+Format examples:
+     {current} / {total}'
+    Result: "3 / 14"
+    '{current} / {total} = {pkg}'
+    Result: "3 / 14 = py3status"
+    '{category} - {pkg}'
+    Result: "x11-misc - py3status"
+
+@author AnwariasEu
+"""
+
+import re
+
+
+class Py3status:
+    """
+    """
+    cache_timeout = 1
+    emerge_log_file = '/var/log/emerge.log'
+    format = u'{current} / {total}'
+    hide_if_zero = True
+
+    def _emergeRunning(self):
+        """
+        Check if emerge is running.
+        Returns true if at least one instance of emerge is running.
+        """
+        try:
+            self.py3.command_output(['pidof', 'emerge'])
+            return True
+        except:
+            return False
+
+    def _getProgress(self):
+        """
+        Get current progress of emerge.
+
+        returns a dict containing current and total value.
+        """
+        ret = {}
+        ret['current'] = None
+        ret['total'] = None
+        ret['err'] = False
+        ret['err_text'] = ""
+        ret['category'] = None
+        ret['pkg'] = None
+
+        input_data = []
+
+        """
+        Try to open file, if unable throw error to be catched later
+        """
+        try:
+            with open(self.emerge_log_file, 'r') as fp:
+                for line in fp:
+                    input_data.append(line)
```

Comment:
Here's an idea. This might be better to use `lines = open(self.log_file).readlines()` and maybe do `[-1]` too for last line only and fix up if/else to work with the last line. Based on the logs, we could make a `status` out of it too... like this: `Cleaning 1/2`, `Emerging 1/2`, `Compiling 1/2`, `Merging 1/2`, `Post-Build Cleaning 1/2`.... if users want that over static `EMG 1/2`... It's a bit more fun. :-)

Reply:
Readlines implemented in 5fc185a.
Status implemented as action in 1d78b05.